blackboxopt/optimization_loops/file_based_distributed.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2024 - for information on the respective copyright owner
+# see the NOTICE file and/or the repository https://github.com/boschresearch/blackboxopt
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import glob
+import json
+import logging
+import time
+from os import PathLike
+from pathlib import Path
+from typing import Callable, List, Optional, Union
+from uuid import uuid4
+
+from blackboxopt import (
+    Evaluation,
+    EvaluationSpecification,
+    Objective,
+    OptimizerNotReady,
+)
+from blackboxopt import logger as default_logger
+from blackboxopt.base import MultiObjectiveOptimizer, SingleObjectiveOptimizer
+from blackboxopt.optimization_loops.utils import (
+    evaluation_function_wrapper,
+    init_max_evaluations_with_limit_logging,
+)
+
+
+def run_optimization_loop(
+    optimizer: Union[SingleObjectiveOptimizer, MultiObjectiveOptimizer],
+    target_directory: PathLike,
+    timeout_s: float = float("inf"),
+    max_evaluations: Optional[int] = None,
+    proposal_queue_size: int = 1,
+    logger: Optional[logging.Logger] = None,
```

Comment:
Should this optimization loop also have the callbacks like the [sequential one](https://github.com/boschresearch/blackboxopt/blob/main/blackboxopt/optimization_loops/sequential.py)? For logging or something similar.